include/drivers/uart.h

```diff
@@ -1127,6 +1139,32 @@ static inline int z_impl_uart_line_ctrl_get(struct device *dev,
 	return -ENOTSUP;
 }
 
+/**
+ * @brief Set the line control callback.
+ *
+ * Sets a callback that will be invoked when the line ctrl state is changed
+ * remotely, such as when the host changes the baud rate on a USB CDC ACM
+ * link.
+ *
+ * @param dev       UART device structure.
+ * @param callback  Event handler.
+ * @param user_data Data to pass to event handler function.
+ */
+__syscall void uart_line_ctrl_callback_set(struct device *dev,
+					   uart_line_ctrl_callback_t callback,
+					   void *user_data);
+
+static inline void z_impl_uart_line_ctrl_callback_set(
+	struct device *dev, uart_line_ctrl_callback_t callback, void *user_data)
+{
+	const struct uart_driver_api *api =
+		(const struct uart_driver_api *)dev->driver_api;
+
+	if (api && api->line_ctrl_callback_set) {
+		api->line_ctrl_callback_set(dev, callback, user_data);
+	}
+}
```

Comment:
Since this API function is optional, it would be good to return `-ENOTSUP` when a given driver does not implement it.

Reply:
Superseded